### ISSUE TITLE
CI testing with different random seeds

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#1741]
#1740 is just one of many issues we've had over the years where something gets committed to master and then shortly thereafter the CI tests start failing intermittently because the new tests aren't resilient to different random seeds.

I thought about it today because I have spent a lot of time over the years debugging problems like this, and I realized we can handle it more automatically.  But what we'll need to do is set up a Jenkins job that can do the following:

 * Look at the diff of a PR to extract any new or modified test cases (these are denoted with `BOOST_AUTO_TEST_CASE()`).
 * Compile `mlpack_test` with some way to set `mlpack::math::RandomSeed(std::time(NULL))` at the beginning of running tests.  Maybe https://www.boost.org/doc/libs/1_60_0/libs/test/doc/html/boost_test/tests_organization/fixtures/global.html is the way to go here.
 * Run each of those modified test cases 1000 times, and output this into an .xml file that can later be parsed by Jenkins to print any failing tests.

We should do this as a bash script or something (I guess Python is fine too) that can be run from the inside of a Jenkins job, so once a script works, we can add it to the jenkins-conf/ repository and then I can set it up inside of Jenkins.  (In fact, go ahead and open a PR there---but I've opened the issue here so it gets some visibility.)

Basically, for this to work right, as input you will get a git repository, which you can then compare with (or rebase on) the mlpack git master branch, and as output, we should get a big unified Boost.Test XML that contains the results of 1000 runs of any new or modified tests that Jenkins can parse.

You can see how Jenkins currently runs the mlpack tests here:

https://github.com/mlpack/jenkins-conf/blob/master/runTestBins.sh

That should help with the options used to get output in the right format.

**Comment by imskr** [comment]
I am beginner!  Can I take this issue?

**Comment by Aakash1822** [comment]
@rcurtin interested to work on this. Please guide me accordingly.

**Comment by rcurtin** [comment]
You're more than welcome to work on this.  I think all of the information that you need should be in the issue description.

**Comment by favre49** [comment]
Considering we have since moved to Azure Pipelines, could we use that instead, and have the script print straight to console? Something like:
`Out of 1000 tests, 21 failed`
We could also print the seeds of the tests that failed. Is any other info required?

**Comment by rcurtin** [comment]
I don't know if we have enough compute time there---1000 runs can often take quite a while (especially if we are running many tests, more than just one).  So maybe Jenkins is the way to go.  In any case, it's easy to plug it into a CI tool once we have a script that runs it.

**Comment by Hardik-Aswal** [comment]
Hey @rcutin I am a Second year student from IIT . I dont have much opensource contribution but I would like to start from here. Im interested to work on it kindly guide me through it
